public/app/plugins/panel/graph/event_editor.ts

```diff
@@ -31,6 +32,16 @@ export class EventEditorCtrl {
     this.timeFormated = this.panelCtrl.dashboard.formatDate(this.event.time!);
   }
 
+  canDelete() {
+    if (contextSrv.accessControlEnabled()) {
+      if (this.event.source.type === 'dashboard') {
```

Comment:
That type refers to the type of the annotations datasource. We can't use `dashboardId` here because it always has the number of the dashboard where the annotation is being rendered. For `organization` annotations we use a `annotation datasource` filtered by tags and the type is not defined for those so this should be safe. Also, this is the legacy component with angular I think we can leave it like this as it is deprecated.

Reply:
Gotcha, thanks for the explanation!